docs/api/rest_api_reference/input/examples/views/POST/ViewInput.json.example

```diff
@@ -0,0 +1,16 @@
+{
+    "ViewInput": {
+        "identifier": "TitleView",
+        "ContentQuery": {
+            "Filter": {
+                "ContentTypeIdentifierCriterion": "image",
+                "SectionIdentifierCriterion": "media"
+            },
+            "limit": 10,
+            "offset": 0,
+            "SortClauses": {
+                "ContentName": "ascending"
+            }
```

Comment:
Can we have the `contentTypeFacetBuilder` like [in the XML example](https://github.com/ezsystems/developer-documentation/blob/add-views-search-sample/docs/api/rest_api_reference/input/examples/views/POST/ViewInput.xml.example#L14)?

Reply:
@Steveb-p

Reply:
So, the thing is, Facet Builders were supposed to be removed in 4.x. While I can construct a query in JSON that creates those facet builder objects internally, I'm not sure if they will work properly.

When reading through the code related to search engine (`Ibexa\Contracts\Core\Repository\SearchService::findContent`) facet builders appear to be unused and therefore silently discarded.

Needs confirmation from @alongosz / @Nattfarinn / @adamwojs: The first two placed the deprecation notice, while Adam is knowledgeable regarding how facet builders / aggregations should behave in context of REST API, or at least could point us in the right direction.

Reply:
> So, the thing is, Facet Builders were supposed to be removed in 4.x. While I can construct a query in JSON that creates those facet builder objects internally, I'm not sure if they will work properly.
> 
> When reading through the code related to search engine (`Ibexa\Contracts\Core\Repository\SearchService::findContent`) facet builders appear to be unused and therefore silently discarded.
> 
> Needs confirmation from @alongosz / @Nattfarinn / @adamwojs: The first two placed the deprecation notice, while Adam is knowledgeable regarding how facet builders / aggregations should behave in context of REST API, or at least could point us in the right direction.

Aggregations should be used instead of Facet builders.

Reply:
@alongosz That is plain. I'll provide concrete questions:
1. Do facet builders still work in context of REST API?
2. Do aggregation work in context of REST API (as I haven't found direct reference to their property in `ViewInput` class yet)?
3. Do you know the structure of XML/JSON that should be used for REST API to make it work (I'm looking for tests that might cover this case and / or directly looking into code at this very moment, but maybe you have some samples on hand)?

Reply:
> @alongosz That is plain. I'll provide concrete questions:
> 
> 1. Do facet builders still work in context of REST API?
> 2. Do aggregation work in context of REST API (as I haven't found direct reference to their property in `ViewInput` class yet)?
> 3. Do you know the structure of XML/JSON that should be used for REST API to make it work (I'm looking for tests that might cover this case and / or directly looking into code at this very moment, but maybe you have some samples on hand)?

It doesn't matter whether it works or not, it shouldn't be advocated in the doc.

To fully answer these questions 1 and 3 I need to spend significant amount of time, so you'll need to wait once I have such time slot available. Question 2 is for Adam, I have no knowledge of Aggregations.

Reply:
@adriendupuis added samples with `Aggregations`, and added a reference to our Search Criteria reference page - to do this I also added markdown dependency on the raml builder tool (which I think would be a nice improvement over trying to squeeze pure HTML into method description, which is also an option).

A few things I've discovered when testing those, so I'll be pushing PRs into rest package, and cross-link them here.